lib/identity_cache/cache_hash.rb

```diff
@@ -1,16 +1,23 @@
-# Use CityHash for fast hashing if it is available; use Digest::MD5 otherwise
-begin
-  require 'cityhash'
-rescue LoadError
+# Use a hashing library for fast hashing if it is available; use Digest::MD5 otherwise
+AVAILABLE_HASHING_GEMS = %w{xxhash murmurhash3 cityhash}
+
+hash_loaded = false
+AVAILABLE_HASHING_GEMS.each do |gem|
```

Comment:
I can see a somewhat contrived scenario when this can be a problem assume you have cityhash in a project and for ...crazy reason you need to add xxhash, it would invalidate the caches without you knowing ?